manager-rs/dauth-service/src/local/queries.rs

```diff
@@ -0,0 +1,955 @@
+use auth_vector::types::Id;
+use sqlx::sqlite::{SqliteConnectOptions, SqlitePool, SqlitePoolOptions, SqliteRow};
+use sqlx::{Sqlite, Transaction};
+
+use crate::data::{database::*, error::DauthError};
+
+/// Constructs the sqlite pool for running queries.
+pub async fn build_pool(database_path: &str) -> Result<SqlitePool, DauthError> {
+    Ok(SqlitePoolOptions::new()
+        .max_connections(10)
+        .connect_with(
+            SqliteConnectOptions::new()
+                .create_if_missing(true)
+                .filename(database_path),
+        )
+        .await?)
+}
+
+/// Creates the auth vector table if it does not exist already.
+pub async fn init_auth_vector_table(pool: &SqlitePool) -> Result<(), DauthError> {
+    sqlx::query(&format!(
+        "CREATE TABLE IF NOT EXISTS {0} (
+            {1} TEXT NOT NULL,
+            {2} INT NOT NULL,
+            {3} BLOB NOT NULL,
+            {4} BLOB NOT NULL,
+            {5} BLOB NOT NULL,
+            PRIMARY KEY ({1}, {2})
+        );",
+        AV_TABLE_NAME, AV_ID_FIELD, AV_RANK_FIELD, AV_XRES_FIELD, AV_AUTN_FIELD, AV_RAND_FIELD
```

Comment:
Is there a reason why these constants were used? Do you envision a use case when they would need to be changed? By creating the queries module, you already are hiding the sqlite queries behind a layer of abstraction, so I'm not sure if there's value in the indirection of using constants here. It makes it a lot harder to reason about what the queries are actually doing. I might be missing something though!

Reply:
The two reasons why I did this are:
1) to allow the names to be changed easier, i.e. we change the value used for something and want to reflect that there. I'm thinking of a case where we query the db outside of rust.
2) To allow quick reference to the names all in one place, instead of searching through the queries file to find them.

I'm not particularly attached to this, and if it's easier to read/more standard to not do this for sql, I can certainly change it. I'm relatively new to writing sql, and my normal programming instinct is to factor out constants like this for easier change and reference.

Reply:
Gotcha! Yeah I think the instinct is right, but in the case of SQL, the schema is closer to an API contract than to an internal variable name than usual. The structure of the queries is actually dependent on the schema, so if you were to change the database schema, you would want to actually double check the queries themselves still made sense. By encoding table and columns names into constants you're saying "my database will always be structured this way", but "at the same time I want to be able to rename anything". Usually it's a better maintenance tradeoff with SQL to use the table and field names directly, since usually if you're changing any of them you're also changing structure, and would want to check your queries anyways.

Re: point 2, it's actually a feature of SQL, not a bug, that the names are distributed! That way when you modify a table, it actually only impacts a subset of queries that were actually using that table. SQLite is intentionally a simple subset of all the SQL optional features so this feels like overkill, but in a larger database with a full-featured DBMS like postgres you have the ability to create things called "views", that seem like tables, but are actually implemented as queries onto other concrete tables (or more views). If there are extra columns or tables in the DBMS that your app doesn't use, that's totally fine : )